app/src/main/java/com/github/studydistractor/sdp/maps/MapsActivity.kt

```diff
@@ -63,27 +64,47 @@ class MapsActivity : AppCompatActivity(), OnMapReadyCallback {
     }
 
     /**
-     * Displays the places on the map and launches the distraction activity when clicking on the marker
+     * Displays a single ProcrastinationActivity on a Google Map and sets up a listener to
+     * open a detailed view of the activity when its associated marker is clicked.
+     *
+     * @param activity The ProcrastinationActivity to display on the map.
      */
-    fun displayPlaces() {
-        val sat = LatLng(46.520544, 6.567825)
-//        TODO: set marker style according to our color scheme and perhaps according to the type of place
-        map?.addMarker(
-            MarkerOptions().position(sat).title("Satellite")
+    private fun displayActivity(activity: ProcrastinationActivity) {
+        val latLng = activity.lat?.let { activity.long?.let { it1 -> LatLng(it, it1) } }
+        val marker = latLng?.let {
+            MarkerOptions()
+                .position(it)
+                .title(activity.name)
                 .icon(BitmapDescriptorFactory.defaultMarker(BitmapDescriptorFactory.HUE_MAGENTA))
-        )
-        map?.setOnInfoWindowClickListener {
-            val activity = ProcrastinationActivity("Satellite", "have a beer at Satellite")
+        }?.let {
+            map?.addMarker(
+                it
+            )
+        }
+        marker?.tag = activity
+        map?.setOnInfoWindowClickListener { activityMarker ->
+            val procActivity = activityMarker.tag as ProcrastinationActivity
             val intent = Intent(this, ProcrastinationActivityActivity::class.java)
-            intent.putExtra("activity", activity)
+            intent.putExtra("activity", procActivity)
             startActivity(intent)
         }
     }
 
+    /**
+     * Displays the places on the map and launches the distraction activity when clicking on the marker
+     */
+    private fun displayPlaces() {
+        FireBaseProcrastinationActivityService().fetchProcrastinationActivities { activities ->
+            for (activity in activities) {
+                displayActivity(activity)
+            }
```

Comment:
The names are misleading I think. The function is called displayPlaces but it uses displayActivity